src/main/java/com/iexec/common/chain/eip712/EIP712Entity.java

```diff
@@ -43,7 +45,7 @@ protected EIP712Entity(EIP712Domain domain, M message) {
 
     @Override
     public HashMap<String, List<TypeParam>> getTypes() {
-        return types;
+        return new HashMap<>(types);
```

Comment:
```suggestion
        return types.clone();
```
Would it be more readable this way ???

Reply:
I prefer to create a new map by hand instead of using the `clone` method. I find it a bit cleaner.